home-assistant/core
# Add uptime sensor to Glances

## Proposed change
This PR adds the uptime sensor to Glances Integration.
It is based on this older, stale PR #95517 with the following updates (which take into account the comments done during the previous PR review):
- update to latest changes on dev branch
- replace custom date parsing by dt.parse_duration
- add check to avoid value flapping (reject small updates to the value)
- update tests to new snapshots method


## Type of change



## Additional information


- This PR fixes or closes issue: 
this PR can replace #95517 and #103168
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/32050

## Checklist



If user exposed functionality or configuration variables are added/changed:


If the code communicates with devices, web services, or third-party tools:

      Updated and included derived files by running: `python3 -m script.hassfest`.
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.



To help with the load of incoming pull requests:


[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure


[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr
